source/fleetprovisioning/FleetProvisioning.cpp

```diff
@@ -523,18 +522,38 @@ bool FleetProvisioning::ProvisionDevice(shared_ptr<SharedCrtResourceManager> fpC
                 DeviceClient::DC_FATAL_ERROR);
             return false;
         }
-        keyPath = config.fleetProvisioning.deviceKey->c_str();
+        if (!config.secureElement.enabled && config.fleetProvisioning.deviceKey.has_value() &&
+            !config.fleetProvisioning.deviceKey->empty())
+        {
+            keyPath = config.fleetProvisioning.deviceKey->c_str();
+        }
+        else
+        {
+            keyPath = "";
+        }
     }
     else
     {
-        if (!CreateCertificateAndKey(identityClient))
+        if (config.secureElement.enabled)
         {
             LOGM_ERROR(
                 TAG,
-                "*** %s: Fleet Provisioning Feature failed to create a new certificate and private key ***",
+                "*** %s: With Secure Element feature is enabled, Device Client only provide support for Fleet "
```

Comment:
Should be either "When Secure Tunneling feature is enabled" or "With Secure Element feature enabled" and then "provides".